solvation_analysis/analysis.py

```diff
@@ -0,0 +1,342 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+
+from MDAnalysis.analysis.base import AnalysisBase
+from MDAnalysis.analysis.rdf import InterRDF
+from MDAnalysis.lib.distances import capped_distance
+import numpy as np
+from solvation_analysis.rdf_parser import identify_solvation_cutoff
+from solvation_analysis.analysis_library import (
+    Coordination,
+    Pairing,
+    Speciation,
+)
+from solvation_analysis.solvation import get_radial_shell, get_closest_n_mol, get_atom_group
+
+
+class Solution(AnalysisBase):
+    """
+    The core class of the solvation module.
+
+    Parameters
+    ----------
+        solute : AtomGroup
+            the solute in the solutions
+        solvents: dict
+            a dictionary of names and atom groups. e.g. {"name_1": solvent_group_1,
+            "name_2": solvent_group_2, ...}
+        radii : dict, optional
+            an optional dictionary of solvation radii, any radii not
+            given will be calculated. e.g. {"name_2": radius_2, "name_5": radius_5}
+        rdf_kernel : function, optional
+            this function must take rdf bins and data as input and return
+            a solvation radius as output. e.g. rdf_kernel(bins, data) -> 3.2. By default,
+            the rdf_kernel is solvation_analysis.rdf_parser.identify_solvation_cutoff.
+        kernel_kwargs : dict, optional
+            kwargs passed to rdf_kernel
+        rdf_init_kwargs : dict, optional
+            kwargs passed to inner rdf initialization
```

Comment:
I clarified the language around the rdf kwargs.

Where should rdf be capitalized? I've left it uncapitalized to try to stick to python naming conventions. I could capitalize it in the docs, but I feel having consistent capitalization might be preferable.

Reply:
IMHO if you're referring to the "analysis method" then it should probably be RDF since it's an acronym. If you're referring to a python method/class then use it's actual name (i.e. rdf_kernel is fine but "inner RDF" is probably more sensible)?

Reply:
Makes sense to me, I changed all the docs rdf -> RDF